yurllib/aasa.go

```diff
@@ -276,3 +282,135 @@ func verifyBundleIdentifierIsPresent(content aasaFile, bundleIdentifier string,
 
 	return false
 }
+
+type appleCDNDebugHeaders struct {
+
+	cdnPath string
+	directFileBody string
+	cachedFileBody string
+	cdnStatusCode int
+
+	CDNQuerySuccess bool `json:"CDNQuerySuccess"`
+	NoAppleFailureMessage bool `json:"AppleFailure"`
+	ASACacheUpToDate bool `json:"ASACacheUpToDate"`
+
+	Failure bool `json:"Failure"`
+
+	AppleFailureDetails string `json:"Apple-Failure-Details"` 
+	AppleFailureReason string `json:"Apple-Failure-Reason"` 
+	AppleFrom string `json:"Apple-From"` 
+	AppleTryDirect string `json:"Apple-Try-Direct"` 
+	CacheControl string `json:"Cache-Control"` 
+	ContentLength string `json:"Content-Length"` 
+	Date string `json:"Date"` 
+	Expires string `json:"Expires"` 
+	Via string `json:"Via"` 
+	Age string `json:"Age"` 
+}
+
+
+func readAppleCDNDebugHeaders(cdnResp *http.Response,cdnPath string, directFileBody string) (appleCDNDebugHeaders , error) {
+	result, err := ioutil.ReadAll(cdnResp.Body)
+	cdnBody := string( result)
+
+	debugHeaders := appleCDNDebugHeaders{
+
+		cdnPath : cdnPath ,
+		directFileBody : directFileBody,
+		cachedFileBody : cdnBody,
+		cdnStatusCode :cdnResp.StatusCode,
+	
+		CDNQuerySuccess: cdnResp.StatusCode<=200 && cdnResp.StatusCode<300,
+		NoAppleFailureMessage: cdnResp.Header.Get("Apple-Failure-Details")=="",
+		ASACacheUpToDate:true,
+
+		Failure: err != nil,
+		
+
+		AppleFailureDetails: cdnResp.Header.Get("Apple-Failure-Details"),
+		AppleFailureReason: cdnResp.Header.Get("Apple-Failure-Reason"),
+		AppleFrom: cdnResp.Header.Get("Apple-From"),
+		AppleTryDirect: cdnResp.Header.Get("Apple-Try-Direct"),
+		CacheControl: cdnResp.Header.Get("Cache-Control"),
+		ContentLength: cdnResp.Header.Get("Content-Length"),
+		Date: cdnResp.Header.Get("Date"),
+		Expires: cdnResp.Header.Get("Expires"),
+		Via: cdnResp.Header.Get("Via"),
+		Age: cdnResp.Header.Get("Age"),		
+	}
+
+	return debugHeaders, nil
+}
+
+func getAppleCDNDebugHeaders(ASAdomain string, directFileBody string) (cdnDebugHeaders appleCDNDebugHeaders) {
+	cdnPath := "https://app-site-association.cdn-apple.com/a/v1/" + ASAdomain
+	cdnResp, err := makeRequest(cdnPath)
+
+	cdnDebugHeaders, readErr :=  readAppleCDNDebugHeaders(cdnResp,cdnPath,directFileBody  )
+	
+	if err != nil || readErr != nil {
```

Comment:
Based on line 342, readErr will always be nil. Is there a case where it could be anything else?

Reply:
Fixed - this was supposed to return `err` on line 341 if ioutil.ReadAll failed